sotu/models.py

```diff
@@ -5,7 +7,57 @@
 database = Database()
 
 
+def generate_random_string(length=32):
+    characters = string.lowercase + string.uppercase + string.digits
+    return ''.join(random.choice(characters) for i in range(length))
+
+
 class Entrant(database.Model):
     github_username = peewee.CharField(unique=True)
     name = peewee.CharField()
     email = peewee.CharField(unique=True)
+
+    def invite(self):
+        invitation = Invitation.create(entrant=self, code=generate_random_string())
```

Comment:
do this in a loop, handling the case where it already exists by generating a new code